fs/src/io_uring/memory.rs

```diff
@@ -14,53 +14,25 @@ use {
 // and chunk it in slices of up to 1G each.
 const FIXED_BUFFER_LEN: usize = 1024 * 1024 * 1024;
 
-pub enum LargeBuffer {
-    Vec(Vec<u8>),
-    HugeTable(PageAlignedMemory),
-}
-
-impl Deref for LargeBuffer {
-    type Target = [u8];
-
-    fn deref(&self) -> &Self::Target {
-        match self {
-            Self::Vec(buf) => buf.as_slice(),
-            Self::HugeTable(mem) => mem.deref(),
+/// Allocate memory buffer optimized for io_uring operations, i.e.
+/// using HugeTable when it is available on the host.
+pub fn new_large_buffer(size: usize) -> std::io::Result<PageAlignedMemory> {
+    log::info!("trying to allocate page-aligned buffer of size {}", size);
+    if size > PageAlignedMemory::page_size() {
+        let size = size.next_power_of_two();
+        if let Ok(alloc) = PageAlignedMemory::alloc(size, true) {
+            log::info!("obtained hugetable io_uring buffer (len={size})");
+            return Ok(alloc);
         }
     }
-}
-
-impl DerefMut for LargeBuffer {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        match self {
-            Self::Vec(buf) => buf.as_mut_slice(),
-            Self::HugeTable(mem) => mem.deref_mut(),
-        }
-    }
-}
-
-impl AsMut<[u8]> for LargeBuffer {
-    fn as_mut(&mut self) -> &mut [u8] {
-        match self {
-            Self::Vec(vec) => vec.as_mut_slice(),
-            LargeBuffer::HugeTable(mem) => mem,
-        }
-    }
-}
-
-impl LargeBuffer {
-    /// Allocate memory buffer optimized for io_uring operations, i.e.
-    /// using HugeTable when it is available on the host.
-    pub fn new(size: usize) -> Self {
-        if size > PageAlignedMemory::page_size() {
-            let size = size.next_power_of_two();
-            if let Ok(alloc) = PageAlignedMemory::alloc_huge_table(size) {
-                log::info!("obtained hugetable io_uring buffer (len={size})");
-                return Self::HugeTable(alloc);
-            }
-        }
-        Self::Vec(vec![0; size])
-    }
+    let alloc = PageAlignedMemory::alloc(size, false).map_err(|e| {
+        io::Error::new(
```

Comment:
I think we should still just return `AllocError` here

Reply:
The reason I did this is because there's all the functions that call `new_large_buffer` have the `std::io::Result` return type. So if we don't convert the error type here we will have to convert upstream multiple times. Let me know if I should still revert to returning AllocError here.